base/BUILD.gn

```diff
@@ -1092,6 +1100,7 @@ component("base") {
 
   if (is_starboard) {
     deps -= [ "//base/allocator/partition_allocator:raw_ptr" ]
+    all_dependent_configs += [ ":starboard_config" ]
```

Comment:
I don't see this being done in trunk?

Reply:
It's also in a config in trunk, see `base_public_defines`, line 99 in  base/BUILD.gn.

Reply:
IDK if `public_config` would suffice here over `all_dependent_configs`, and given that `public_configs` only apply to direct dependencies that seems fragile and the related bugs are very hard to debug.

Reply:
We should do `public_configs`. While it may be hard to debug, so is the converse case of this config getting propagated somewhere we don't want it

Reply:
As you can see, with public_configs now many binaires are getting ASAN failures that result from the define missing.

That's what I mean with fragile. The dependents compile, but with a mismatching ABI.

[test_runner.py:789:INFO]   FAILED TESTS GROUPS: dom_test, xhr_test, media_capture_test, webdriver_test, web_test, worker_test, js_profiler_test

Reply:
I propose making this a separate task for someone to followup on.

Reply:
Following up on it sgtm, would you just make a TODO bug and put it in the backlog? (and also put the TODO in //net?)

Reply:
TODO added in https://github.com/youtube/cobalt/pull/2612/commits/ee4fd0c9cbdf45bcb6fcba74af9914d36ef0879b